src/libraries/System.Private.Xml/src/System/Xml/Schema/XmlValueConverter.cs

```diff
@@ -268,6 +269,8 @@ protected XmlBaseConverter(XmlTypeCode typeCode)
             _typeCode = typeCode;
         }
 
+        [UnconditionalSuppressMessage("AotAnalysis", "IL3050:AotUnfriendlyApi",
+                Justification = "All types that are instantiated by XmlValueConverters are used elsewhere in the implementations of the DatatypeImplementation abstract class.")]
```

Comment:
I took a cursory look at the code and I am not able to figure out what guarantees that the array types are available.

Also, you can simplify `_clrTypeDefault = Array.CreateInstance(converterAtomic.DefaultClrType!, 0).GetType();` to `_clrTypeDefault = converterAtomic.DefaultClrType!.MakeArrayType();` while you are on it.

Reply:
I've changed this implementation to more directly assign the array type so we don't need to use `MakeArrayType` here anyway.